.github/workflows/test.yml

```
@@ -48,3 +48,60 @@ jobs:
 
       - name: Run widget tests, unit tests
         run: flutter test --machine --coverage
+
+  generate_mason_bundle:
+    # The job generate_mason_bundle identify that the job test must complete successfully
+    # before this generate_mason_bundle job will run
+    needs: test
+    name: Generate Mason bundle
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    steps:
+      - name: Check out
+        uses: actions/checkout@v3
+
+      - name: Set up Flutter environment
+        uses: subosito/flutter-action@v2
+        with:
+          channel: 'stable'
+          flutter-version: '3.10.5'
+
+      - name: Remove the old sample project
+        run: |
+          rm -rf sample
+
+      - name: Install Mason CLI
+        run: |
+          dart pub global activate mason_cli
+          mason get
+
+      - name: Generate Mason bundle
+        run: |
+          mason bundle bricks/permission_handler -t dart -o bricks/template/hooks/bundles
+
+      - name: Set the project name "sample"
+        uses: jossef/action-set-json-field@v2.1
+        with:
+          file: mason-config.json
+          field: project_name
+          value: sample
+
+      - name: Generate the new sample project
+        run: |
+          mason make template -c mason-config.json
+
+      - id: changes
+        name: Check for changes in the sample project
+        run: |
+          count=$(git status sample --porcelain | wc -l)
+          echo "count=$count" >> $GITHUB_OUTPUT
```

we need to check mason bricks changes

```suggestion
      - id: changes
        name: Check for changes in the mason bricks
        run: |
          count=$(git status bricks --porcelain | wc -l)
          echo "count=$count" >> $GITHUB_OUTPUT
```